tests/mock_cloud_sdk.py

```diff
@@ -0,0 +1,143 @@
+
+        Check how many times this function has been called for a given
+        job and updates its progress accordingly.
+        On the first call, the job status is set to RUNNING.
+        After self.max_progress_steps, the job status is set to DONE and
+        fake results are set.
+
+        Args:
+            id (str): The ID of the job for which progress is being simulated.
+        """
+        progress_step = self.jobs_progress_counter[id]
+        if progress_step == 0:
+            self.jobs[id].status = "RUNNING"
+        if progress_step >= self.max_progress_steps:
+            self.jobs[id].status = "DONE"
+            self.jobs[id]._full_result = {"counter": {"1001": 50, "1011": 25}, "raw": []}
+        self.jobs_progress_counter[id] += 1
+
+
+class MockSDK:
+    """Helper class to mock the cloud SDK and skip the API calls.
+
+    Warning: This implements only a small subset of the functions available
+        in the cloud SDK; focusing on the functions used by qek.
+        This should be extended to support all methods and moved to the
+        pasqal-cloud repository so that it can be reused for all future
+        libraries using the SDK.
+    """
+
+    def __init__(self) -> None:
+        self.mock_server = MockServer()
+
+    def get_device_specs_dict(self) -> Any:
+        """Retrieve the device specifications from a local JSON file."""
+        with open("tests/fixtures/device_specs.json", "r") as f:
+            return json.load(f)
+
+    def create_batch(
+        self,
+        serialized_sequence: str,
+        jobs: list[CreateJob],
+        open: bool | None = None,
+        emulator: EmulatorType | None = None,
+        configuration: BaseConfig | None = None,
+        wait: bool = False,
+    ) -> Batch:
+        """Create a batch of jobs and simulate its creation in the mock server."""
+        batch_id = str(uuid4())
+        batch = Batch(
+            id=batch_id,
+            open=bool(open),
+            complete=bool(open),
+            created_at="",
+            updated_at="",
+            device_type=emulator if emulator else "FRESNEL",
+            project_id="",
+            user_id="",
+            status="DONE",
+            jobs=[
+                {
+                    **j,
+                    "batch_id": batch_id,
+                    "id": str(uuid4()),
+                    "project_id": "",
+                    "status": "PENDING",
+                    "created_at": "",
+                    "updated_at": "",
+                }
+                for j in jobs
+            ],
+            configuration=configuration,
+            _client=MagicMock(),
+        )
+
+        self.mock_server.set_job(batch.ordered_jobs[0])
+        return batch
+
+    def get_jobs(self, filters: JobFilters) -> PaginatedResponse:
+        """Retrieve jobs based on filters, simulating the SDK's get_jobs call."""
+        items: list[Job] = []
+
+        # TODO: support other cases
```

Comment:
Nit: Could you detail what you mean by "other cases"?

Reply:
I detailed the comment in the code